### PR TITLE
Consider all known runs when checking whether versions are verified

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review. Gets rid of a sad-face you put in a comment a long time ago :)

### CODE REVIEW

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/maintenance/UpgraderTest.java [hunk 1]
```diff
@@ -1034,21 +1033,9 @@ public void testsEachUpgradeCombinationWithFailingDeployments() {
...
-        // Need to test v2->v3 combination again before upgrading second deployment (not really, but this is a limitation of lastSuccess)
-        tester.triggerJobs();
-        assertEquals(v2, application.instance().deploymentJobs().jobStatus().get(stagingTest).lastTriggered().get().sourcePlatform().get());
-        assertEquals(v3, application.instance().deploymentJobs().jobStatus().get(stagingTest).lastTriggered().get().platform());
-        application.runJob(stagingTest);
-
         // Second deployment upgrades
         application.runJob(productionUsWest1);
 
-        // ... now we have to test v1->v3 again :(
```
Comment (jonmv) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/maintenance/UpgraderTest.java]: 🔫

**Comment by jonmv** [comment]
Test failure, hang on.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/MetricsReporter.java [hunk 2]
```diff
@@ -142,47 +139,42 @@ private int nodesFailingOsUpgrade() {
...
                            .reduce(Duration::plus)
                            .map(totalDuration -> totalDuration.dividedBy(jobDurations.size()))
                            .orElse(Duration.ZERO);
     }
 
-    private static Map<ApplicationId, Integer> deploymentWarnings(List<Instance> instances) {
-        return instances.stream()
+    private static Map<ApplicationId, Integer> deploymentWarnings(DeploymentStatusList statuses) {
+        return statuses.asList().stream()
+                       .flatMap(status -> status.application().instances().values().stream())
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/MetricsReporter.java]: Nit: Alignment.

**Review by mpolden — APPROVED** [review]
Great!

**Pull request merged**